src/components/coin-row/FastRequestCoinRow.tsx

```diff
@@ -0,0 +1,143 @@
+    justifyContent: 'space-between',
+  },
+  column: {
+    flex: 1,
+    justifyContent: 'space-between',
+    marginLeft: 11,
+  },
+  icon: {
+    justifyContent: 'center',
+  },
+  topRow: {
+    flexDirection: 'row',
+  },
+  wholeRow: {
+    flexDirection: 'row',
+    height: 40,
+    justifyContent: 'space-between',
+    marginVertical: 17,
+    paddingHorizontal: 19,
+  },
+});
+
+const getPercentageOfTimeElapsed = (startDate: Date, endDate: Date) => {
+  const originalDifference = differenceInMinutes(endDate, startDate);
+  const currentDifference = differenceInMinutes(endDate, Date.now());
+
+  return Math.floor((currentDifference * 100) / originalDifference);
+};
+
+export default React.memo(function RequestCoinRow({
+  item,
+  theme,
+}: {
+  item: any;
+  theme: ThemeContextProps;
+}) {
+  const buttonRef = useRef();
+  const dispatch = useDispatch();
+  const { navigate } = useNavigation();
+
+  const [state, setState] = useState<{
+    expirationColor: string;
+    expiresAt: Date | null;
+    percentElapsed: number;
+  }>({
+    expirationColor: '',
+    expiresAt: null,
+    percentElapsed: 0,
+  });
+
+  const { colors } = theme;
+
+  const minutes =
+    state.expiresAt && differenceInMinutes(state.expiresAt, Date.now());
+
+  useEffect(() => {
+    if (item?.displayDetails?.timestampInMs) {
+      const createdAt = new Date(item.displayDetails.timestampInMs);
+      const expiresAt = addHours(createdAt, 1);
+      const percentElapsed = getPercentageOfTimeElapsed(createdAt, expiresAt);
+
+      setState({
+        expirationColor: percentElapsed > 25 ? colors.appleBlue : colors.orange,
+        expiresAt,
+        percentElapsed,
+      });
+    }
+  }, [colors, item]);
+
+  const handleExpiredRequests = useCallback(() => {
+    if (state.expiresAt && isPast(state.expiresAt)) {
+      dispatch(removeRequest(item.requestId));
+    }
+  }, [dispatch, state.expiresAt, item.requestId]);
+
+  const handlePressOpen = useCallback(() => {
+    navigate(Routes.CONFIRM_REQUEST, {
+      transactionDetails: item,
+    });
+  }, [item, navigate]);
+
+  useEffect(() => {
+    handleExpiredRequests();
+  }, [state.expiresAt, handleExpiredRequests]);
+
+  return (
+    <ButtonPressAnimation
+      onPress={handlePressOpen}
+      scaleTo={0.98}
+      waitFor={buttonRef}
+    >
+      <View style={cx.wholeRow}>
+        <View style={cx.icon}>
+          <FastRequestCoinIcon
+            dappName={item.dappName}
+            expirationColor={state.expirationColor}
+            imageUrl={item.imageUrl}
+            percentElapsed={state.percentElapsed}
+            theme={theme}
+          />
+        </View>
+        <View style={cx.column}>
+          <View style={cx.topRow}>
+            <Text
+              color={{ custom: state.expirationColor }}
+              size="14px"
+              weight="semibold"
+            >
+              {lang.t('exchange.coin_row.expires_in', {
+                minutes: minutes || 0,
```

Comment:
Why not use `??`

Reply:
I guess it doesn't change much here. The only difference would be that `NaN` would be changed to `0`, which wouldn't happen with `??`. So maybe makes sense to leave it as it was?